modules/sample-http4s/src/test/scala/core/Http4s/Http4sDebugBodyTest.scala

```diff
@@ -0,0 +1,74 @@
+package core.Http4s
+
+import _root_.debugBody.client.{ http4s => generatedClient }
+import _root_.debugBody.server.http4s._
+import _root_.debugBody.server.http4s.definitions._
+import cats.effect.IO
+import cats.effect.IO._
+import org.http4s.client.Client
+import org.http4s.implicits._
+import org.scalatest.EitherValues
+import org.scalatest.funsuite.AnyFunSuite
+import org.scalatest.matchers.should.Matchers
+import io.circe._
+import org.http4s._
+import org.http4s.syntax._
+import org.http4s.circe.CirceInstances
+import cats.syntax.all._
+import cats.effect.syntax.all._
+import fs2._
+
+class Http4sDebugBodyTest extends AnyFunSuite with Matchers with EitherValues with CirceInstances {
+
+  private val server = new Resource[IO]()
+    .routes(
+      new Handler[IO] {
+        override def debugBody(respond: DebugBodyResponse.type)(body: Body): IO[DebugBodyResponse] =
+          IO.pure(respond.NoContent)
+      }
+    )
+    .orNotFound
+
+  private def sendRequest(server: HttpApp[IO], body: Json) = {
+    val req = Request[IO](
+      method = Method.POST,
+      uri = Uri.unsafeFromString("http://localhost:1234/bar"),
+      body = jsonEncoderOf[IO, Json].toEntity(body).body
+    )
+    Client.fromHttpApp[IO](server).run(req)
+  }
+
+  test("should fail with a helpful error message") {
+    val invalidJson = Json.obj(
+      ("invalidSomething1") -> Json.fromInt(1),
+      ("something2")        -> Json.fromString("something")
+    )
+    val request        = sendRequest(server, invalidJson)
+    val actualResponse = request.use(_.pure[IO]).unsafeRunSync()
+    val actualErrorMessages =
+      Stream
+        .resource(request)
+        .flatMap(_.bodyText)
+        .compile
+        .toList
+        .unsafeRunSync()
+
+    val expectedErrorMessage =
+      "The request body was malformed. Underlying message: Attempt to decode value on failed cursor. Problematic fields: something1"
```

Comment:
Since you have functionality for multiple errors reported, would you mind adding a third test for when both `something1` and `something2` were invalid? If that's desired to be supported (I think so), it would be good to have a test for it.

Otherwise, just surfacing the raw exception from circe, to ensure we get the most information from the underlying exception.

Reply:
I think that'd require swapping `jsonOf` to `accumulatingJsonOf` when generating decoders to make it work and I actually did at first but couldn't extract all the errors at once for some reason, it was always just the first one.

I'll try once again and report back 👍

Reply:
No worries, thanks for taking a look.

If you do get it working, I think concatenating each `toString` of each of the validation errors, which would likely give more context for each error, one per line.